Stackup-Rwanda/knights-bn-frontend
# #170947607 Users should be able to view and edit their profiles

#### What does this PR do?

- Users should be able to view and edit their profiles.

#### Description of Task to be completed?

- allow the user to view his profile.
- allow a user to edit his profile.
- test the endpoint.

#### How should this be manually tested?
- run `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git` to clone the repo.
- run `cd knights-bn-backend` in your terminal to enter into the repository.
- run `git fetch ` command to download commits, files, and refs from a remote repository into your local repo.
- run `git ft-view-and-edit-profile-170947607` to switch to this branch.
- run `npm install` to install dependencies.
- run `npm run dev-start` to start the application.
  you will see the login page then login with your email or password or log in using your social account `Google & Facebook` 
- use `http://localhost:8080/profile` to access your profile where you can edit anything you want including your Profile picture.

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#170947607](https://www.pivotaltracker.com/story/show/170947607)

#### Screenshots (if appropriate)

        Login page. 



<img width="1437" alt="Screen Shot 2020-05-12 at 17 41 21" src="https://user-images.githubusercontent.com/50768983/81714875-d859bc00-9477-11ea-92a2-cb332d9cd372.png">

      View Profile Image
<img width="1440" alt="Screen Shot 2020-05-12 at 17 16 42" src="https://user-images.githubusercontent.com/50768983/81714947-edcee600-9477-11ea-8560-1e312890ef47.png">

     Update Profile Image by choosing a different image location.

<img width="1440" alt="Screen Shot 2020-05-12 at 17 45 23" src="https://user-images.githubusercontent.com/50768983/81715406-806f8500-9478-11ea-91da-9e1ffdfb2890.png">

     Resize the chosen Profile image then crop

<img width="1439" alt="Screen Shot 2020-05-12 at 17 45 40" src="https://user-images.githubusercontent.com/50768983/81715559-a7c65200-9478-11ea-9138-d7866bbe2ced.png">

    Updated Profile image 

<img width="1440" alt="Screen Shot 2020-05-12 at 17 44 45" src="https://user-images.githubusercontent.com/50768983/81715670-c6c4e400-9478-11ea-8048-4dd5ccd2d7b7.png">


- You can also change different displayed information.
- Click on update to confirm your change 
- You will see a message that says `Profile updated successfully`
- You cancel your changes also by clicking on cancel 

#### Questions:
N/A
